### DISCUSSION TITLE
How to pass entirety of conf dict to BashOperators?

### DISCUSSION THREAD

**Discussion by jeffbraun-8451** [apache/airflow#26120] [Q&A]
Hello, I have a DAG that is triggered via a HTTP request that is accompanied by a conf dictionary. The contents of that dictionary are subject to change, and all of my DAG tasks are BashOperators that kick off python scripts that utilize the data found in the conf dictionary. Is there a way to pass the entire conf dictionary to the BashOperator instead of breaking them out via their keys with jinja templating?

**Comment by potiuk** [comment]
There is a doc for that https://airflow.apache.org/docs/apache-airflow/stable/howto/operator/bash.html#templating

**Reply by potiuk** [reply]
you need to break it by keys.

**Reply by potiuk** [reply]
BTW. There is no differnce of "whole context" vs. keys. You will have to refer through keys anyway as  there is no "typed" structure to pass to bash - so whatever your refer to in bash, it has to be some form of string name of parameter you have to access. There is no other way to access any fields in Bash. So I am not sure what you want to achieve.
